File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    answer = []
    s = set()
    l = []
    for index, i in enumerate(words) :
        if not s and not l :
            l.append(i)
            s.add(i)
            continue
        if i in s :
            return [index % n + 1,  index // n + 1]
        else :
            if l[-1][-1] == i[0] :
                l.append(i)
                s.add(i)
            else :
                return [index % n + 1,  index // n + 1]
    else :
        return [0, 0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;

class Solution {
    public static int[] solution(int n, String[] words) {
        List<String> list = new ArrayList<>();

        for (int i = 0; i < words.length; i++) {
            if (list.contains(words[i])) {
                return new int[]{((i + 1) % n == 0) ? n : (i + 1) % n, i / n + 1};
            }

            if (i > 0 && words[i - 1].charAt(words[i - 1].length() - 1) != words[i].charAt(0)) {
                return new int[]{((i + 1) % n == 0) ? n : (i + 1) % n, i / n + 1};
            }

            list.add(words[i]);
        }

        return new int[]{0, 0};
    }
}